.github/workflows/deploy_doc.yml

```diff
@@ -4,11 +4,45 @@ name: Deploy docs
 on: 
   workflow_dispatch:
   push:
-      branches: [master]
+      branches: [master,versions] # TODO: remove branch versions 
+      tags:
   
 jobs:
+  evaluate-label:
+    runs-on: ubuntu-latest
+    outputs:
+      label: ${{ steps.label_step.outputs.version}}
+    steps:
+      - run: echo ${{ github.event_name }} #TODO: remove this debugging line 
+      - run: echo ${{ github.ref }} # TODO: remove this debugging line 
+      - name: checks for the label 
+        id: label_step 
+        run: |
+          if [[ "${{ github.event_name }}" == "push" ]] && [[ "${{ github.ref  }}" == "refs/heads/versions" ]]; then
+                echo "version=latest" >> $GITHUB_OUTPUT
+          fi
+          if [[ "${{ github.event_name }}" == "push" ]] && [[ "${{ github.ref  }}" != "refs/heads/versions" ]]; then
+                echo "version=Error" >> $GITHUB_OUTPUT
+          fi
+          if [[ "${{ github.event_name }}" == "tag" ]]; then
```

Comment:
Tag is scoped under `push` in the workflow file, are you sure the `github.event_name` would be `tag`?